frontend/pages/policies/ManagePoliciesPage/components/ManageAutomationsModal/ManageAutomationsModal.tsx

```diff
@@ -0,0 +1,230 @@
+import React, { useState } from "react";
+
+import Modal from "components/Modal";
+import Button from "components/buttons/Button";
+// @ts-ignore
+import Checkbox from "components/forms/fields/Checkbox";
+// @ts-ignore
+import InputField from "components/forms/fields/InputField";
+import IconToolTip from "components/IconToolTip";
+
+import { IPolicy } from "interfaces/policy";
+import { IWebhookFailingPolicies } from "interfaces/webhook";
+import { useDeepEffect } from "utilities/hooks";
+import { size } from "lodash";
+
+import PreviewPayloadModal from "../PreviewPayloadModal";
+
+interface IManageAutomationsModalProps {
+  onCancel: () => void;
+  onCreateAutomationsSubmit: (formData: IWebhookFailingPolicies) => void;
+  togglePreviewPayloadModal: () => void;
+  showPreviewPayloadModal: boolean;
+  availablePolicies: IPolicy[];
+  currentAutomatedPolicies?: number[];
+  currentDestinationUrl?: string;
+}
+
+interface ICheckedPolicy {
+  name?: string;
+  id: number;
+  isChecked: boolean;
+}
+
+const useCheckboxListStateManagement = (
+  allPolicies: IPolicy[],
+  automatedPolicies: number[] | undefined
+) => {
+  const [policyItems, setPolicyItems] = useState<ICheckedPolicy[]>(() => {
+    return (
+      allPolicies &&
+      allPolicies.map((policy) => {
+        return {
+          name: policy.name,
+          id: policy.id,
+          isChecked:
+            !!automatedPolicies && automatedPolicies.includes(policy.id),
+        };
+      })
+    );
+  });
+
+  const updatePolicyItems = (policyId: number) => {
+    setPolicyItems((prevState) => {
+      const selectedPolicy = policyItems.find(
+        (policy) => policy.id === policyId
+      );
+
+      const updatedPolicy = selectedPolicy && {
+        ...selectedPolicy,
+        isChecked: !!selectedPolicy && !selectedPolicy.isChecked,
+      };
+
+      // this is replacing the policy object with the updatedPolicy we just created.
+      const newState = prevState.map((currentPolicy) => {
+        return currentPolicy.id === policyId && updatedPolicy
+          ? updatedPolicy
+          : currentPolicy;
+      });
+      return newState;
+    });
+  };
+
+  return { policyItems, updatePolicyItems };
+};
+
+const validateAutomationURL = (url: string) => {
+  const errors: { [key: string]: string } = {};
+
+  if (!url) {
+    errors.url = "Please add a valid destination URL";
+  }
+
+  const valid = !size(errors);
+  return { valid, errors };
+};
```

Comment:
We should validate that the URL is a valid URL. If we don't have a `isValidURL()` function in our utility library we should add one. Looks like `utilities/strings/stringUtils.ts` would be a good spot. Here's a good regex: 

```
function isValidURL(str) {
  var pattern = new RegExp('^(https?:\\/\\/)?'+ // protocol
    '((([a-z\\d]([a-z\\d-]*[a-z\\d])*)\\.)+[a-z]{2,}|'+ // domain name
    '((\\d{1,3}\\.){3}\\d{1,3}))'+ // OR ip (v4) address
    '(\\:\\d+)?(\\/[-a-z\\d%_.~+]*)*'+ // port and path
    '(\\?[;&a-z\\d%_.~+=-]*)?'+ // query string
    '(\\#[-a-z\\d_]*)?$','i'); // fragment locator
  return !!pattern.test(str);
}
```

Reply:
Ok, I started this issue https://github.com/fleetdm/fleet/issues/3507 because as I built the failing policies webhook, I followed the frontend pattern already for host status webhook URL validator, which only tested if a string existed.